bindings/python/scripts/generate.py

```diff
@@ -199,6 +199,9 @@ def handle_struct_decl(self):
         else:
             self.linelist.append(f'py::class_<{class_name}>(m, "{class_name}")')
 
+        # default constructor
+        self.linelist.append(".def(py::init<>())")
```

Comment:
Would this cause issues if there's already a ctor before or if default ctor is deleted?

Reply:
Handled via an if condition: don't add default ctor in `handle_constructor`
```diff
-        self.linelist.append(f".def(py::init<{argument_type_list}>())")
+        # default ctor `.def(py::init<>())` already inserted while handling struct/class decl
+        if argument_type_list:
+            self.linelist.append(f".def(py::init<{argument_type_list}>())")
```